src/vite.ts

```diff
@@ -68,6 +68,24 @@ async function bundle (nuxt: Nuxt, builder: any) {
 
   await ctx.nuxt.callHook('vite:extend', ctx)
 
+  ctx.nuxt.hook('vite:serverCreated', async ({ server }: { server: vite.ViteDevServer }) => {
+    const warmedUrls: string[] = []
+    const normalizeResult = (url: string) => server.transformRequest(url).then(r => typeof r === 'string' ? r : r.code)
+    const processScript = async (html: string) => {
+      const results = html.matchAll(/^import.*'(.*)'$/mg)
```

Comment:
I think we can load graph from viteServer api directly and directly for `/client.ts` entrypoint

Reply:
I believe the graph has to be populated by the incoming requests rather than having been calculated already by vite.

Would be good to get a performance test up so we can wee what the difference is between processing `/`, `/client.js` or `/@vite/client`. I'll see if I can do some local benchmarking.

Reply:
It doesn't make a huge difference in terms of what URLs end up getting downloaded, but `/client.js` seems to get somewhat better coverage (51 vs 48 files warmed up) 👍

As far as benchmarking, this does not yet make a huge difference in terms of loading time as measured by devtools time to fully loaded, but I will chew over it.

Possibly related: not all files are getting cached in the dependency tree (e.g 16 out of 42 on one initial fixture load weren't loaded from cache). In another project, 96/124 were not preloaded according to the below code, even when the same URL had been called by `server.transformRequest`.

*Example code*:
```js
    viteServer.moduleGraph.getModuleByUrl(originalURL).then((mod) => {
      const cached = mod.ssrTransformResult || mod.transformResult
      console.log(originalURL, !!cached)
    })
```

Reply:
> It doesn't make a huge difference in terms of what URLs end up getting downloaded

By purpose it does :) We might use heavy dependencies on index chunk and development of `/faq` page does not need them to be warmed up. Only `/client.js` entry is shared for sure and going to be used always

Reply:
Yes, that's what I'm working with locally, just haven't pushed yet.

Reply:
> Avoid using html parsing

Do you mean avoiding parsing the JS?

Reply:
Yes. The variable was called `html` :D

Reply:
My fault! Let me go back to the drawing board if parsing the JS for import strings isn't okay. We might need to use a rollup plugin or similar 🤔